src/libraries/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -573,7 +574,7 @@ public static XmlSerializer[] FromMappings(XmlMapping[] mappings, Type type)
             {
                 XmlSerializer[] serializers = new XmlSerializer[mappings.Length];
                 for (int i = 0; i < serializers.Length; i++)
-                    serializers[i] = (XmlSerializer)contract.TypedSerializers[mappings[i].Key];
+                    serializers[i] = (XmlSerializer)contract!.TypedSerializers[mappings[i].Key!]!;
```

Comment:
So before nullability was enabled, `serializers` could accept `null`. With this change, if `contract!.TypedSerializers[mappings[i].Key!]` returns `null`, this would cause a NRE?

Reply:
nullability annotations won't cause NRE anywhere, it might produce some unexpected result for people seeing method/property marked as non-null but actually getting null. I'll double check if this is correctly annotated

Reply:
yes, this is fine - mappings contains a list of known entries in the Hashtable and Hashtable always returns nullable as it's not strongly typed but if it was a Dictionary then value would be non nullable

Reply:
Hmmm, maybe it won't cause an NRE but if `contract!.TypedSerializers[mappings[i].Key!]` returns a null`, won't the cast to `(XmlSerializer)` fail now?

Reply:
no, it will just end up with null even though it's typed as non-null